daemon/cmd/daemon_main.go

```diff
@@ -1679,7 +1702,8 @@ func runDaemon(ctx context.Context, cleaner *daemonCleanup, shutdowner hive.Shut
 	}
 	bootstrapStats.k8sInit.End(true)
 	restoreComplete := d.initRestore(restoredEndpoints)
-	if wgAgent != nil {
+
+	if wgAgent, ok := d.datapath.WireguardAgent().(*wireguard.Agent); ok && wgAgent != nil {
```

Comment:
Rather than casting, can you add whatever methods are missing to the interface?

Reply:
That will create an import cycle (ipcache depends on pkg/datapath, Init() takes *IPCache). I'll instead pass *wg.Agent down as a separate param.